addons/website_event_track/static/src/js/service_worker.js

```diff
@@ -236,14 +236,21 @@ const matchCache = async (request) => {
 
 /**
  *
- * @param {FetchEvent} param0
+ * @param {Request} request
+ * @param {object} [options]
+ *      {Boolean} disableTracking whether adding a header preventing the server to track the request (default: false)
  * @returns {Promise<Response>}
  */
-const processFetchRequest = async ({ request }) => {
+const processFetchRequest = async (request, options) => {
+    options = options || {};
     const requestCopy = request.clone();
     let response;
     try {
-        response = await fetch(request);
+        if (options.disableTracking) {
```

Comment:
`if (options && options.disableTracking)` to allow calling the function without proving options since you marked it as optional in the docstring?

Reply:
Actually, I had added the following statement at the beginning of the function:
options = options || {};

But as there is only one argument, I have used your solution.